src/Utils.js

```diff
@@ -222,6 +222,11 @@ export function withSetStateAllowed(fn) {
   fn();
   if (cleanup) {
     delete global.document;
+    // This works around a bug in node/jest in that developers aren't able to
```

Comment:
can this be done *before* deleting it, so that it's always set to undefined, and then deleted when possible?